syntax = "proto3";

package transport_catalogue;

import "map_renderer.proto";

import "transport_router.proto";

message Stop {
    string name = 1;
    double latitude = 2;
    double longitude = 3;
}

message Bus {
    string name = 1;
    bool is_ring = 2;
}

message Distance {
    uint32 start_stop = 1;
    uint32 finish_stop = 2;
    uint32 distance = 3;
}

message Route {
    uint32 bus_name = 1;
    repeated uint32 stop_names = 2;
}

message Catalogue {
    repeated Stop stops = 1;
    repeated Bus buses = 2;
    repeated Distance distance_info = 3;
    repeated Route route_info = 4;
    Map_renderer.MapRenderer map_renderer = 5;
    Router.RouterSetting router_setting = 6;
}